src/telemetry/events.ts

```diff
@@ -19,6 +19,14 @@ export enum UserEvent {
   CCloudAuthentication = "CCloud Authentication",
   ViewSearchAction = "View Search Action",
   SchemaAction = "Schema Action",
+  ToolUsage = "Tool Usage",
+  ToolInvocation = "Tool Invocation",
+  ToolInvocationError = "Tool Invocation Error",
+  ToolInvocationSuccess = "Tool Invocation Success",
+  ToolInvocationCancelled = "Tool Invocation Cancelled",
+  ToolInvocationTimeout = "Tool Invocation Timeout",
+  ToolInvocationFailure = "Tool Invocation Failure",
+  ToolInvocationWarning = "Tool Invocation Warning",
```

Comment:
One thing to be careful about here is that each event type will create its own table, so we should aim to use a single event and provide additional data with a `status` property. Maybe just `ChatToolUsage`?